datanode/sqlstore/migrations/0040_volume_discount_programs.sql

```diff
@@ -0,0 +1,59 @@
+-- +goose Up
+
+-- create a new volume discount record when a new volume discount program is created,
+-- updated, or ended so that we keep an audit trail, just in case.
+-- We create it as a hypertable and set a retention policy to make sure
+-- old and redundant data is removed in due course.
+create table if not exists volume_discount_programs
+(
+    id                       bytea                    not null,
+    version                  int                      not null,
+    benefit_tiers            jsonb,
+    end_of_program_timestamp timestamp with time zone not null,
+    window_length            int                      not null,
+    vega_time                timestamp with time zone not null,
+    ended_at                 timestamp with time zone,
+    primary key (vega_time)
```

Comment:
Should this be `(vega_time, seq_num)` to avoid the same bug we saw with the referrals program where a volume discounted program start/ends on the same block?

Reply:
We can use the version, that should do the trick